src/main/java/com/treasure/hunt/view/ToolbarController.java

```diff
@@ -1,36 +1,55 @@
 package com.treasure.hunt.view;
 
-import javafx.fxml.FXML;
+import javafx.beans.binding.BooleanBinding;
 import javafx.scene.Group;
 import javafx.scene.control.Label;
 import javafx.scene.control.ToggleButton;
 import javafx.scene.control.ToggleGroup;
 import javafx.scene.layout.Pane;
 import lombok.Getter;
+import lombok.extern.slf4j.Slf4j;
 
 /**
+ * The toolbar controller holds two sections for buttons.
+ * These buttons show and hide widgets in the corresponding section in the {@link WidgetBarController}.
+ *
  * @author jotoh
  */
+@Slf4j
 public class ToolbarController {
-    @FXML
-    private Pane toolbar;
+    public Pane first;
+    public Pane second;
     @Getter
-    private ToggleGroup toggleGroup = new ToggleGroup();
+    private ToggleGroup firstGroup = new ToggleGroup();
+    @Getter
+    private ToggleGroup secondGroup = new ToggleGroup();
 
-    public void initialize() {
-    }
+    public void addButton(boolean first, String text, boolean selected, Pane widget) {
+
+        if (widget == null) {
+            throw new NullPointerException("Widget pane mustn't be null.");
```

Comment:
or maybe just return, throwing null pointers seems weird to me, u normally dont do this